src/main/java/htsjdk/samtools/util/output/async/AsyncBlockCompressedOutputStream.java

```diff
@@ -0,0 +1,317 @@
+package htsjdk.samtools.util.output.async;
+
+import htsjdk.samtools.util.BlockCompressedStreamConstants;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.samtools.util.RuntimeIOException;
+import htsjdk.samtools.util.output.AbstractBlockCompressedOutputStream;
+import htsjdk.samtools.util.zip.DeflaterFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.*;
+import java.util.function.Supplier;
+import java.util.zip.CRC32;
+import java.util.zip.Deflater;
+
+public class AsyncBlockCompressedOutputStream extends AbstractBlockCompressedOutputStream {
+
+    private static final int COMPRESSING_THREADS = countCompressingThread();
+
+    private static final ExecutorService gzipExecutorService = Executors.newFixedThreadPool(COMPRESSING_THREADS, runnable -> {
+        Thread thread = Executors.defaultThreadFactory().newThread(runnable);
+        thread.setDaemon(true);
```

Comment:
Good to have thread name configuration in here. Will look better in log files 
thread.setName("GZIP-Thread");